### ISSUE TITLE
maxpool outside

### ISSUE THREAD

**Issue by johnnydept** [AlexeyAB/darknet#451]
Hi, the following cfg is from tiny-yolo-voc.cfg. 
How does the maxpool output get from 13x13 to 13x13 using size=2, stride=1? 
The formula is OutWidth=(Width−Filter size)/Stride+1 , im getting 12x12 
`[maxpool]
size=2
stride=1`

**Comment by AlexeyAB** [comment]
Hi,

https://github.com/AlexeyAB/darknet/blob/15c89e7a714e7e37c13618eace9325a06f0642fc/src/maxpool_layer.c#L30-L31

input: `w = 13`

output: `l.out_w = (w + 2*padding)/stride` = `(13 + 2 x 0) / 1 = 13/1 = 13`

----

Darknet Tiny-yolo requires specifying a special mode of max-pooling `pad_mode = SAME` in terms of Tensorflow: https://github.com/opencv/opencv/blob/0792ef87893af0946611d45a57e73a63f12dcb9a/modules/dnn/src/darknet/darknet_io.cpp#L197

What is the difference between 'SAME' and 'VALID' padding in tf.nn.max_pool of tensorflow? 
 https://stackoverflow.com/questions/37674306/what-is-the-difference-between-same-and-valid-padding-in-tf-nn-max-pool-of-t

**Comment by johnnydept** [comment]
Once again, Thank you Alex :)
